apps/minifront/src/components/shared/asset-selector.tsx

```diff
@@ -46,8 +45,7 @@ const switchAssetIfNecessary = ({
   }
 };
 
-const useFilteredAssets = ({ value, onChange, filter }: AssetSelectorProps) => {
-  const { assets } = useLoaderData() as SwapLoaderResponse;
```

Comment:
`<AssetSelector />` is a shared component (i.e., it's in the `shared` directory), so it shouldn't use `useLoaderData()`, since that means it can only be used when a parent component has a specific loader data structure. Instead, I've modified `<AssetSelector />` to accept an `assets` prop that it now passes to `useFilteredAssets`.